### PR TITLE
CMake

### PR DESCRIPTION
**Pull request by adriandavila**
This PR accomplishes:
- Use CMake as buildsystem generator
- Should hopefully make it easier to add new dependencies (e.g. a logger) going forward
- Should enable us to automatically pull dependencies in the future once I have a better handle of cmake

### CODE REVIEW

File: client/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+# Create the executable target
+add_executable(client 
+    client.cpp
+    ../host/redis.cpp 
+    ../crypto/encryption_engine.cpp
+)
+
+# # Set compiler flags
+# target_compile_options(client PRIVATE -g ${CXXFLAGS})
```
Comment (adriandavila) [inline client/CMakeLists.txt:10]: will remove hanging comment

File: .gitignore
Comment (jasondu7297) [inline .gitignore:1]: We should remove any redundant gitignore files in the subdirectories.

**Comment by jasondu7297** [comment]
As per comment here: https://github.com/ySteinhart1/ORTOA/pull/6#discussion_r1359679991, was `#pragma once` able to be used?

**Comment by adriandavila** [comment]
> As per comment here: [#6 (comment)](https://github.com/ySteinhart1/ORTOA/pull/6#discussion_r1359679991), was `#pragma once` able to be used?

I briefly tested it and it seems to work fine. I'm not going to make that change in this PR. Here's some relevant discussion about it if you're curious
https://stackoverflow.com/questions/787533/is-pragma-once-a-safe-include-guard

**Pull request merged**